tests/framework/extensions/workloads/create.go

```diff
@@ -0,0 +1,69 @@
+package workloads
+
+import (
+	stevev1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/services"
+	"github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	containerNamePrefix = "tcon"
+	wlNamePrefix        = "twl"
+	defaultNamespace    = "default"
+	isCattleLabeled     = true
+	serviceNamePrefix   = "svc-"
+)
+
+func CreatSimpleDeployment(steveclient *stevev1.Client, containerImage string) (*stevev1.SteveAPIObject, error) {
+	containerName := namegenerator.AppendRandomString(containerNamePrefix)
+	containerTemplate := NewContainer(containerName, containerImage, corev1.PullAlways, []corev1.VolumeMount{}, []corev1.EnvFromSource{}, nil, nil, nil)
+
+	podTemplate := NewPodTemplate([]corev1.Container{containerTemplate}, []corev1.Volume{}, []corev1.LocalObjectReference{}, nil)
+
+	wlName := namegenerator.AppendRandomString(wlNamePrefix)
+	deployment := NewDeploymentTemplate(wlName, defaultNamespace, podTemplate, isCattleLabeled, nil)
+
+	deploymentResp, err := steveclient.SteveType(DeploymentSteveType).Create(deployment)
+	if err != nil {
+		return nil, err
+	}
+
+	return deploymentResp, err
+}
+
+func CreateService(steveclient *stevev1.Client, serviceNameSuffix string, servicePort int32, serviceType corev1.ServiceType, selector map[string]string) (*stevev1.SteveAPIObject, error) {
+	service := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceNamePrefix + serviceNameSuffix,
+			Namespace: defaultNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{
+				{
+					Name: "port",
+					Port: servicePort,
+				},
+			},
+			Selector: selector,
+		},
+	}
+	serviceResp, err := steveclient.SteveType("service").Create(service)
+	if err != nil {
+		return nil, err
+	}
+	return serviceResp, nil
+}
```

Comment:
Don't we have the template already in `services/template.go` in the extensions? The same comment that I made above, this is limiting the create call and forcing users to use this in my opinion.

Actually, I think even having this template also proves my point, if we have such small actions capsulated, we also indirectly ask users to look in the codebase, and it falls on familiarity. 

We should aim for minimum familiarity and user-friendly generic helpers in the extensions in my opinion. 

Templates are flexible in terms of these;

- We can add later a field that doesn't exist in a template arg - because it's returning the k8s object (in Anupama's latest addition for jobs)
- We can choose not to use the template at all (in monitoring test private helper)

But the create call wrappers are not because this is introducing another cluster create problem that we had before, and the only way to manipulate that "created object" becomes adding more args to those calls